### PR TITLE
TC #2 - New Prescription Flow Frontend Setup

### PR DESCRIPTION
**Pull request by pecezon**
## Summary
This pull request creates the base content for the doctors section to create new prescriptions, it makes use of tabs to control the flow.

### Search Patient By Id
The first but basic functional part of making a new prescription was selecting the patient, to select a patient you will need a unique id and validate it after putting it in.

### Projection For Next PR Regarding The New Prescription Flow
I have planned to make a new prescription in 3 main steps:

1. Selecting the patient.
2. Selecting the drugs with their corresponding instructions of use.
3. Confirming the final prescriptions

This 3 steps are contemplated by the tab navigation added to the new component `NewPrescriptionBody`

## Testing
There was not a lot to test but the functionality of the search patient by ID flow, this was successfully implemented taking care of the edge cases (Invalid Patient Id Provided) as seen in the following video:

<div>
    <a href="https://www.loom.com/share/7a8ea6542be74747b9fbfe2405bdd476">
      <p>Select Patient By Id - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/7a8ea6542be74747b9fbfe2405bdd476">
      
    </a>
  </div>

### CODE REVIEW

File: frontend/src/api/userFunctions.js
Comment (ysong4) [inline frontend/src/api/userFunctions.js:1]: nit: maybe remove the 'Function' in the file name. Rename it to something like 'users.js'

File: frontend/src/api/userFunctions.js [hunk 1]
```diff
@@ -20,12 +20,28 @@ async function fetchUserId(email) {
     }
   );
   if (!response.ok) {
-    throw new Error("Failed to fetch doctor's name");
+    throw new Error("Failed to fetch user by email");
+  }
+
+  const app = await response.json();
```
Comment (ysong4) [inline frontend/src/api/userFunctions.js:26]: nit: rename app -> userData

File: frontend/src/components/doctor/PatientSelection.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import { NumberInput, Button, addToast } from "@heroui/react";
+import { fetchUserById } from "../../api/userFunctions";
+import PatientInfo from "./PatientInfo";
+import PastPrescriptions from "./PastPrescriptions";
+import PastMedicalRecords from "./PastMedicalRecords";
+
+export default function NewPrescriptionMain() {
```
Comment (ysong4) [inline frontend/src/components/doctor/PatientSelection.jsx:8]: Rename to something else? For example, following the file name, PatientSelection

File: frontend/src/components/doctor/PatientSelection.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  const [curPatientId, setCurPatientId] = useState();
+  const [selectedPatient, setSelectedPatient] = useState();
```
Comment (ysong4) [inline frontend/src/components/doctor/PatientSelection.jsx:10]: nit: rename it to [curPatientInfo, setCurPatientInfo]

**Review by ysong4 — APPROVED** [review]
Overall LGTM!
